### ISSUE TITLE
ERROR WHEN CASTING 

### ISSUE THREAD

**Issue by trode21** [kevinatown/MMM-Screencast#26]
The raspberry pi comes up on my YouTube phone app and when i try and cast it this error comes up please help!

screencast stderr: /home/pi/MagicMirror2/modules/MMM-Screencast/node_modules/electron/dist/electron: /lib/arm-linux-gnueabihf/libc.so.6: version `GLIBC_2.27' not found (required by /home/pi/MagicMirror2/modules/MMM-Screencast/node_modules/electron/dist/electron)

screencast stderr: npm ERR! code ELIFECYCLE
npm
screencast stderr:  ERR! errno 1

screencast stderr: npm ERR! magicmirror-screencast@0.0.3 start: `electron app.js "http://www.youtube.com/tv?pairingCode=0125953A-8FA4-479E-9779-31B029AD5D54&rUrl=ws%3A%2F%2F192.168.50.251%3A53785%2FA24BF7D9-088C-427C-84CB-B044B7BEA6A4&rId=58E7B5E1-9CD0-4127-A840-0D9F8427EAEA&theme=cl&launch=dial-watch" "bottomRight" "500" "300"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the magicmirror-screencast@0.0.3 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

screencast stderr: 

screencast stderr: npm ERR! A complete log of this run can be found in:
npm ERR!     /home/pi/.npm/_logs/2019-01-03T06_11_50_936Z-debug.log

**Comment by mqm165** [comment]
I am receiving the same error 

I've tried on multiple iOS devices with he latest updates, I am using MM on Raspberry pi 3 B+ with 2018-11-13-raspbian-stretch-full and latest update of Node js, NPM, MM2, PM2.

I've tried removing all the modules from the config file and leave only the screen cast, still didn't work.

Following is the log file for the error:




0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node',
1 verbose cli   '/usr/bin/npm',
1 verbose cli   'start',
1 verbose cli   'http://www.youtube.com/tv?pairingCode=BE1A3639-F6FA-454B-80BE-DE13BE3906DC&rUrl=ws%3A%2F%2F192.168.8.170%3A61998%2F3AFCE519-E4B4-4AC4-BA38-6B1EDF211784&rId=6055A92C-FD49-4DEA-8A13-AD83B009439C&theme=cl&launch=dial-watch',
1 verbose cli   'bottomRight',
1 verbose cli   '500',
1 verbose cli   '300' ]
2 info using npm@6.4.1
3 info using node@v10.15.0
4 verbose run-script [ 'prestart', 'start', 'poststart' ]
5 info lifecycle magicmirror-screencast@0.0.3~prestart: magicmirror-screencast@0.0.3
6 info lifecycle magicmirror-screencast@0.0.3~start: magicmirror-screencast@0.0.3
7 verbose lifecycle magicmirror-screencast@0.0.3~start: unsafe-perm in lifecycle true
8 verbose lifecycle magicmirror-screencast@0.0.3~start: PATH: /usr/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/pi/MagicMirror/modules/MMM-Screencast/node_modules/.bin:/usr/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/pi/MagicMirror/node_modules/.bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/games:/usr/games
9 verbose lifecycle magicmirror-screencast@0.0.3~start: CWD: /home/pi/MagicMirror/modules/MMM-Screencast
10 silly lifecycle magicmirror-screencast@0.0.3~start: Args: [ '-c',
10 silly lifecycle   'electron app.js "http://www.youtube.com/tv?pairingCode=BE1A3639-F6FA-454B-80BE-DE13BE3906DC&rUrl=ws%3A%2F%2F192.168.8.170%3A61998%2F3AFCE519-E4B4-4AC4-BA38-6B1EDF211784&rId=6055A92C-FD49-4DEA-8A13-AD83B009439C&theme=cl&launch=dial-watch" "bottomRight" "500" "300"' ]
11 silly lifecycle magicmirror-screencast@0.0.3~start: Returned: code: 1  signal: null
12 info lifecycle magicmirror-screencast@0.0.3~start: Failed to exec start script
13 verbose stack Error: magicmirror-screencast@0.0.3 start: `electron app.js "http://www.youtube.com/tv?pairingCode=BE1A3639-F6FA-454B-80BE-DE13BE3906DC&rUrl=ws%3A%2F%2F192.168.8.170%3A61998%2F3AFCE519-E4B4-4AC4-BA38-6B1EDF211784&rId=6055A92C-FD49-4DEA-8A13-AD83B009439C&theme=cl&launch=dial-watch" "bottomRight" "500" "300"`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:301:16)
13 verbose stack     at EventEmitter.emit (events.js:182:13)
13 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:182:13)
13 verbose stack     at maybeClose (internal/child_process.js:962:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:251:5)
14 verbose pkgid magicmirror-screencast@0.0.3
15 verbose cwd /home/pi/MagicMirror/modules/MMM-Screencast
16 verbose Linux 4.14.79-v7+
17 verbose argv "/usr/bin/node" "/usr/bin/npm" "start" "http://www.youtube.com/tv?pairingCode=BE1A3639-F6FA-454B-80BE-DE13BE3906DC&rUrl=ws%3A%2F%2F192.168.8.170%3A61998%2F3AFCE519-E4B4-4AC4-BA38-6B1EDF211784&rId=6055A92C-FD49-4DEA-8A13-AD83B009439C&theme=cl&launch=dial-watch" "bottomRight" "500" "300"
18 verbose node v10.15.0
19 verbose npm  v6.4.1
20 error code ELIFECYCLE
21 error errno 1
22 error magicmirror-screencast@0.0.3 start: `electron app.js "http://www.youtube.com/tv?pairingCode=BE1A3639-F6FA-454B-80BE-DE13BE3906DC&rUrl=ws%3A%2F%2F192.168.8.170%3A61998%2F3AFCE519-E4B4-4AC4-BA38-6B1EDF211784&rId=6055A92C-FD49-4DEA-8A13-AD83B009439C&theme=cl&launch=dial-watch" "bottomRight" "500" "300"`
22 error Exit status 1
23 error Failed at the magicmirror-screencast@0.0.3 start script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]

**Comment by kevinatown** [comment]
@trode21 did you follow the setup steps for MMM-Screencast? Your error looks like there might be an issue with the electron version you are using. Pretty much you need to (from the MM base directory):

```bash
cd modules/MMM-Screencast
npm i
```
Then go back to the base directory of and start it again.

@trode21 and @mqm165 couple other questions:
- Where in the directory structure is the MMM-Screencast kept? is it just in `modules` folder or is it nested?
- Can you provide more information from the entire MM startup?

**Comment by trode21** [comment]
Yes my Screencast folder is in the modules folder and that is where i installed it as well. I got it to come up on my phone but it will not connect and play the video.

**Comment by kevinatown** [comment]
Hmm... 

@trode21 So i have some other questions:
- have you tried removing all of the node_modules for MMM-Screencast and reinstalling them:
```bash
cd modules/MMM-Screencast
rm -rf node_modules
npm i
```
- Are you using any other bistro besides raspbian on your pi?
- Do you have an external sound card/how are you connecting sound to your pi?
- Is there anything else that could be using this sound card?

Sorry, that I have been kinda absent, been swamped with work.

**Comment by AdnanAhmed97** [comment]
An easy way would be rebuiding the electron

**Comment by Hakawe** [comment]
Hey man,
I have got a similar issue and can't figure out why it doesn't work.
INFO: I am an absolute beginner with raspian, etc. I just want to set up my MM and only used some manuals for it. And my english is pretty bad, so I am sorry if its hard to follow.


What happened:
**console:**

<

pi@raspberrypi:~ $ cd MagicMirror
pi@raspberrypi:~/MagicMirror $ npm start

> magicmirror@2.6.0 start /home/pi/MagicMirror
> sh run-start.sh

getrandom indicates that the entropy pool has not been initialized. Rather than continue with poor entropy, this process will block until entropy is available.
Starting MagicMirror: v2.6.0
Loading config ...
Loading module helpers ...
No helper found for module: alert.
Initializing new module helper ...
Module helper loaded: updatenotification
No helper found for module: clock.
Initializing new module helper ...
Module helper loaded: calendar
Initializing new module helper ...
Module helper loaded: MMM-voice
Initializing new module helper ...
Module helper loaded: MMM-TTS
Initializing new module helper ...
Module helper loaded: MMM-Screencast
No helper found for module: compliments.
No helper found for module: currentweather.
No helper found for module: weatherforecast.
Initializing new module helper ...
Module helper loaded: newsfeed
All module helpers loaded.
Starting server on port 8080 ... 
Server started ...
Connecting socket for: updatenotification
Connecting socket for: calendar
Starting node helper for: calendar
Connecting socket for: MMM-voice
Starting module helper: MMM-voice
Connecting socket for: MMM-TTS
Starting node helper for: MMM-TTS
Connecting socket for: MMM-Screencast
Connecting socket for: newsfeed
Starting module: newsfeed
Sockets connected & modules started ...
Launching application.
Create new calendar fetcher for url: https://www.feiertage-deutschland.de/kalender-download/force-download.php - Interval: 300000
Create new news fetcher for url: http://www.tagesschau.de/xml/rss2 - Interval: 300000
MMM-voice: Checking files.
DIAL Server is running on PORT 8569
MMM-voice: Starting pocketsphinx.
screencast stdout: 
> magicmirror-screencast@0.0.3 start /home/pi/MagicMirror/modules/MMM-Screencast
> electron app.js "http://www.youtube.com/tv?pairingCode=e0661e3d-0a5d-412c-a344-f5bdd08af78f&theme=cl&dialLaunch=watch" "topCenter" "500" "300"


screencast stderr: /home/pi/MagicMirror/modules/MMM-Screencast/node_modules/electron/index.js:9
  throw new Error('Electron failed to install correctly, please delete node_modules/electron and try installing again')
  ^

Error: Electron failed to install correctly, please delete node_modules/electron and try installing again
    at Object.<anonymous> (/home/pi/MagicMirror/modules/MMM-Screencast/node_modules/electron/index.js:9:9)
    at Module._compile (internal/modules/cjs/loader.js:701:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:712:10)
    at Module.load (internal/modules/cjs/loader.js:600:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:539:12)
    at Function.Module._load (internal/modules/cjs/loader.js:531:3)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
    at Object.<anonymous> (/home/pi/MagicMirror/modules/MMM-Screencast/node_modules/electron/cli.js:3:16)
    at Module._compile (internal/modules/cjs/loader.js:701:30)

screencast stderr: npm ERR! code ELIFECYCLE
npm ERR! errno 1

screencast stderr: npm ERR!
screencast stderr:  magicmirror-screencast@0.0.3 start: `electron app.js "http://www.youtube.com/tv?pairingCode=e0661e3d-0a5d-412c-a344-f5bdd08af78f&theme=cl&dialLaunch=watch" "topCenter" "500" "300"`
npm ERR! Exit status 1

screencast stderr: npm ERR! 
npm ERR!
screencast stderr:  Failed at the magicmirror-screencast@0.0.3 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

screencast stderr: 

screencast stderr: npm ERR! A complete log of this run can be found in:
npm ERR!     /home/pi/.npm/_logs/2019-03-23T20_00_24_046Z-debug.log

closing code: 1

<
<
<

 **Here the named ERR_log_file:**
[2019-03-23T20_00_24_046Z-debug.log](https://github.com/kevinatown/MMM-Screencast/files/2999778/2019-03-23T20_00_24_046Z-debug.log)

<
<

I already tried some things, like:

**cd modules/MMM-Screencast
rm -rf node_modules
npm i**

or delete the named electron-folder
and reinstalled
<
<

didn't work for me...
also tried to stream from different phones, didn't work either.

I hope you can help me.

**Comment by aHVzY2g** [comment]
+1 I got the same error running in servermode with a fresh installtion.
```npm ERR!     /home/mspangenberg/.npm/_logs/2019-03-25T12_40_21_528Z-debug.log

closing code: 1
^C[SIGINT] Received. Shutting down server...
Stopping module helper: updatenotification
Stopping module helper: MMM-Screencast
Stopping module helper: calendar
Stopping module helper: newsfeed
mspangenberg@Fhem:~/MMSCREEN/MagicMirror$ cat  /home/mspangenberg/.npm/_logs/2019-03-25T12_40_21_528Z-debug.log
0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node',
1 verbose cli   '/usr/bin/npm',
1 verbose cli   'start',
1 verbose cli   'http://www.youtube.com/tv?pairingCode=xxx&theme=cl&dialLaunch=watch',
1 verbose cli   'bottomRight',
1 verbose cli   '500',
1 verbose cli   '300' ]
2 info using npm@6.4.1
3 info using node@v10.15.3
4 verbose run-script [ 'prestart', 'start', 'poststart' ]
5 info lifecycle magicmirror-screencast@0.0.3~prestart: magicmirror-screencast@0.0.3
6 info lifecycle magicmirror-screencast@0.0.3~start: magicmirror-screencast@0.0.3
7 verbose lifecycle magicmirror-screencast@0.0.3~start: unsafe-perm in lifecycle true
8 verbose lifecycle magicmirror-screencast@0.0.3~start: PATH: /usr/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/mspangenberg/MMSCREEN/MagicMirror/modules/MMM-Screencast/node_modules/.bin:/home/mspangenberg/bin:/home/mspangenberg/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
9 verbose lifecycle magicmirror-screencast@0.0.3~start: CWD: /home/mspangenberg/MMSCREEN/MagicMirror/modules/MMM-Screencast
10 silly lifecycle magicmirror-screencast@0.0.3~start: Args: [ '-c',
10 silly lifecycle   'electron app.js "http://www.youtube.com/tv?pairingCode=0b1a6119-18c3-4e4a-a968-1235527ba6de&theme=cl&dialLaunch=watch" "bottomRight" "500" "300"' ]
11 silly lifecycle magicmirror-screencast@0.0.3~start: Returned: code: 1  signal: null
12 info lifecycle magicmirror-screencast@0.0.3~start: Failed to exec start script
13 verbose stack Error: magicmirror-screencast@0.0.3 start: `electron app.js "http://www.youtube.com/tv?pairingCode=xxx&theme=cl&dialLaunch=watch" "bottomRight" "500" "300"`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:301:16)
13 verbose stack     at EventEmitter.emit (events.js:189:13)
13 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:189:13)
13 verbose stack     at maybeClose (internal/child_process.js:970:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:259:5)
14 verbose pkgid magicmirror-screencast@0.0.3
15 verbose cwd /home/mspangenberg/MMSCREEN/MagicMirror/modules/MMM-Screencast
16 verbose Linux 4.4.0-143-generic
17 verbose argv "/usr/bin/node" "/usr/bin/npm" "start" "http://www.youtube.com/tv?pairingCode=xxx&theme=cl&dialLaunch=watch" "bottomRight" "500" "300"
18 verbose node v10.15.3
19 verbose npm  v6.4.1
20 error code ELIFECYCLE
21 error errno 1
22 error magicmirror-screencast@0.0.3 start: `electron app.js "http://www.youtube.com/tv?pairingCode=xxx&theme=cl&dialLaunch=watch" "bottomRight" "500" "300"`
22 error Exit status 1
23 error Failed at the magicmirror-screencast@0.0.3 start script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]
```

Using the youtube paring url from the log, I was able to stream to it openend in firefox.

**Comment by Hakawe** [comment]
Problem solved!

Just run electron's install.js manually. Somehow `npm install` didn't do its job fully (also did update electron's dependency to version 2.0.16, perhaps the combination did it).
